src/Viewer.tsx

```diff
@@ -832,17 +833,27 @@ function Viewer(): ReactElement {
   // the frame using a debounced value to prevent constant updates as it moves.
   const debouncedFrameInput = useDebounce(frameInput, 250);
   useEffect(() => {
-    setFrame(debouncedFrameInput);
+    if (!timeControls.isPlaying() && currentFrame !== debouncedFrameInput) {
+      setFrame(debouncedFrameInput);
+    }
```

Comment:
clarifying q: the time slider gaining focus will pause playback, but this `if` protects against this effect still firing once or twice before that happens?

Reply:
I found it useful to prevent a stuttering I was seeing, where if playback is enabled and a frame takes more than 250 ms to load, `debouncedFrameInput` would update and attempt to set the frame to the current one, which could cause a race condition with the next frame being loaded. `setFrame` will temporarily pause playback and resume it once the frame is loaded, which led to a lot of fighting as `timeControls` was rapidly paused/unpaused.